app/client/websocket/index.ts

```diff
@@ -165,34 +160,35 @@ export default class WebSocketClient {
                 this.sendMessage('authentication_challenge', {token: this.token});
             }
 
-            if (this.connectFailCount > 0) {
+            if (this.shouldSkipSync) {
+                logInfo('websocket connected to', this.url);
+                this.firstConnectCallback?.();
+            } else {
                 logInfo('websocket re-established connection to', this.url);
                 if (!reliableWebSockets && this.reconnectCallback) {
                     this.reconnectCallback();
                 } else if (reliableWebSockets) {
```

Comment:
Then we do we need realiableWebsockets still?

Reply:
Some of the cases handled here are...
- First connect
  - The init logic handles the sync
  - The init logic starts the websockets with "shouldSkipSync" set to true
  - The websocket just connects and call the first connect callback
- A network hiccup disconnects the websocket without reliable websockets
  - The websockets closes, and onClose reopens the connection after a while (and makes sure shouldSkipSync is false)
  - In onOpen, since we should not skip the sync, and there are not reliable websockets, we do the full reconnect.
- A network hiccup disconnects the websocket with reliable websockets, and is able to continue the connection
  - The websockets closes, and onClose reopens the connection after a while (and makes sure shouldSkipSync is false)
  - In onOpen, since we should not skip the sync and we have reliable websockets, we just call the reliableReconnectCallback, and continue
  - When it finishes connecting, we will receive the hello message. The connection id will be the same, and the sequence will match, so no sync needed.
- A network hiccup disconnects the websocket with reliable websockets, and is able to continue the connection but there is a sequence break
  - The websockets closes, and onClose reopens the connection after a while (and makes sure shouldSkipSync is false)
  - In onOpen, since we should not skip the sync and we have reliable websockets, we just call the reliableReconnectCallback, and continue
  - When it finishes connecting, we will receive the hello message. The connection id is the same, but the sequence don't match. We close the connection and clear the connection id.
  - onClose restarts the connection, and onOpen just calls the `reliableReconnectCallback`
  - We receive a new hello message. The connection id won't match, so we do a full sync.